src/App.jsx

```diff
@@ -207,7 +217,32 @@ function App() {
                           }`}
                         >
                           <ErrorBoundary inApp>
-                            {posts.map(({ id, post }) => (
+                          <div
+                            // className="search-bar"
+                            
+                          >
+                            <input
+                              type="search"
+                              style={{background:'white'}}
+                              // className="search-input" 
+                              value={searchText}
+                              placeholder="Search Here..."
+                              onChange={handleSearch}
+                            />
+                          </div>
+                            {searchText ? posts.filter((post) => post.post.username?.toLowerCase().includes(searchText.toLowerCase()))
```

Comment:
this will just search in the current paginated feed, would that be okay? Or we should do something else like querying from actual db? What do you think?

Reply:
I didn't get it @narayan954 can you please be more specific

Reply:
in the searchbutton you're just filtering the posts array. (post array is paginated and contains just 10 posts initially) Filtering this array could only filter on the current posts you're seeing.

Reply:
Aren't the posts coming from actual db

Reply:
@narayan954  is pagination enabled in the project?

Reply:
Yes, but in batches, like 10 at a time, so in our app, there're only 10 posts loaded. If we search, we should be able to search from global instead of just these 10 posts right?

Reply:
Okay

Reply:
@narayan954 I am liitle bit confused do you want search results among these 10 posts or you want all the posts related to the searched user ? Because the logic i have given they are going to search among the posts those are visible on screen

Reply:
> @narayan954 I am liitle bit confused do you want search results among these 10 posts or you want all the posts related to the searched user ? Because the logic i have given they are going to search among the posts those are visible on screen

The correct implementation is getting all related posts to a user, I was just explaining to you that your logic is just filtering on these 10 posts data :)